src/index.ts

```diff
@@ -16,92 +16,118 @@
+export * from "./analysis/static/targetUtil";
+
+// - Graph
+export * from "./analysis/static/graph/drawGraph";
+
+// Event System
+export * from "./event/EventManager";
+export * from "./event/ListenerInterface";
+export * from "./event/ProgramState";
+
+// Factories
+export * from "./factories/CrossoverFactory";
+export * from "./factories/ObjectiveManagerFactory";
+export * from "./factories/SamplerFactory";
+export * from "./factories/SearchAlgorithmFactory";
+export * from "./factories/TerminationFactory";
+export * from "./factories/UserInterfaceFactory";
+
+// Plugin
+export * from "./plugin/CrossoverPlugin";
+export * from "./plugin/ListenerPlugin";
+export * from "./plugin/ObjectiveManagerPlugin";
+export * from "./plugin/PluginInterface";
+export * from "./plugin/PluginManager";
+export * from "./plugin/SamplerPlugin";
+export * from "./plugin/SearchAlgorithmPlugin";
+export * from "./plugin/TerminationPlugin";
+export * from "./plugin/UserInterfacePlugin";
 
 // Search
-export * from "./search/SearchSubject";
-export * from "./search/EncodingSampler";
-export * from "./search/Encoding";
-export * from "./search/ExecutionResult";
 export * from "./search/Archive";
+export * from "./search/Decoder";
+export * from "./search/Encoding";
 export * from "./search/EncodingRunner";
+export * from "./search/EncodingSampler";
+export * from "./search/ExecutionResult";
+export * from "./search/SearchListener";
+export * from "./search/SearchSubject";
 
-// Budget
+// - Budget
+export * from "./search/budget/Budget";
+export * from "./search/budget/BudgetListener";
 export * from "./search/budget/BudgetManager";
 export * from "./search/budget/EvaluationBudget";
 export * from "./search/budget/IterationBudget";
 export * from "./search/budget/SearchTimeBudget";
+export * from "./search/budget/StagnationBudget";
 export * from "./search/budget/TotalTimeBudget";
 
-// Termination
-export * from "./search/termination/TerminationManager";
-export * from "./search/termination/SignalTerminationTrigger";
+// - Comparators
+export * from "./search/comparators/DominanceComparator";
 
-// Statistics
-export * from "./statistics/CoverageWriter";
-export * from "./statistics/RuntimeVariable";
-export * from "./statistics/StatisticsCollector";
-export * from "./statistics/StatisticsSearchListener";
-export * from "./statistics/SummaryWriter";
+// - Metaheuristics
+export * from "./search/metaheuristics/RandomSearch";
+export * from "./search/metaheuristics/SearchAlgorithm";
 
-// Factories
-export * from "./search/factories/AlgorithmFactory";
-export * from "./search/factories/TerminationFactory";
+export * from "./search/metaheuristics/evolutionary/EvolutionaryAlgorithm";
+export * from "./search/metaheuristics/evolutionary/NSGAII";
+export * from "./search/metaheuristics/evolutionary/Sfuzz";
 
-// Objective
-export * from "./search/objective/ObjectiveFunction";
+export * from "./search/metaheuristics/evolutionary/MOSAFamily";
```

Comment:
Why are the search algorithms separated with new lines?

Reply:
Fixed